### PR TITLE
Add OSP github action

### PR DESCRIPTION
**Pull request by julek-wolfssl**

### CODE REVIEW

**Comment by julek-wolfssl** [comment]
Depends on https://github.com/wolfSSL/osp/pull/133

**Comment by julek-wolfssl** [comment]
Now depends on https://github.com/julek-wolfssl/wolfssl-actions.

File: .github/workflows/osp-check.yml [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+            --with-crypto-library=wolfssl 
+            WOLFSSL_CFLAGS="-I$GITHUB_WORKSPACE/build-dir/include/ -I$GITHUB_WORKSPACE/build-dir/include/wolfssl"
+            WOLFSSL_LIBS="-L$GITHUB_WORKSPACE/build-dir/lib -lwolfssl"
+          check: true
+
+# TODO: Currently this test fails. Enable it once it becomes passing.
+#  haproxy_check:
+#    strategy:
+#      matrix:
+#        os: [ ubuntu-latest ]
```
[the remark this answers is not in the document]
Comment (julek-wolfssl) [reply .github/workflows/osp-check.yml:78]: Removed.

File: .github/workflows/osp-check.yml [hunk 2]
```diff
@@ -0,0 +1,118 @@
+name: stunnel Tests
+
+on:
+  workflow_call:
+
+jobs:
+  stunnel_check:
+    strategy:
+      matrix:
+        os: [ ubuntu-latest ]
```
[the remark this answers is not in the document]
Comment (julek-wolfssl) [reply .github/workflows/osp-check.yml:10]: Removed.

File: .github/workflows/osp-check.yml [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+          check: true
+
+      - name: Confirm stunnel built with wolfSSL
+        working-directory: ./stunnel
+        run: ldd src/stunnel | grep wolfssl
+
+  openvpn_check:
+    strategy:
+      matrix:
+        os: [ ubuntu-latest ]
```
[the remark this answers is not in the document]
Comment (julek-wolfssl) [reply .github/workflows/osp-check.yml]: Removed.

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -10,11 +10,13 @@ on:
...
-    call_multi-compiler:
+    multi-compiler:
         uses: ./.github/workflows/multi-compiler.yml
-    call_docker-OpenWrt:
+    docker-OpenWrt:
         uses: ./.github/workflows/docker-OpenWrt.yml
-    call_os-check:
+    os-check:
         uses: ./.github/workflows/os-check.yml
+    osp:
```
[the remark this answers is not in the document]
Comment (julek-wolfssl) [reply .github/workflows/main.yml]: Please take a look again. Now the tests are in separate files.

File: .github/workflows/haproxy-check.yml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+on:
+  workflow_call:
+
+jobs:
+  haproxy_check:
+    strategy:
+      matrix:
+        # List of refs to test
+        ref: [ master ]
```
Comment (bandi13) [inline .github/workflows/haproxy-check.yml:11]: You don't actually use this variable.

**Review by bandi13 — APPROVED** [review]
Looks good to me! Nice work.

**Comment by bandi13** [comment]
> Looks good to me! Nice work.

After the `make user_settings.h` test is fixed of course.

**Comment by julek-wolfssl** [comment]
That should be the `user_settings.h` fixed.

**Pull request merged**